core/src/main/scala/cats/data/Validated.scala

```diff
@@ -1035,6 +1042,15 @@ sealed abstract private[data] class ValidatedInstances2 {
   // scalastyle:off method.length
 }
 
+private[data] class ValidatedSelective[E: Semigroup] extends ValidatedApplicative[E] with Selective[Validated[E, *]] {
+  override def select[A, B](fab: Validated[E, Either[A, B]])(ff: Validated[E, A => B]): Validated[E, B] =
```

Comment:
Are there any laws that relate select to ap? I worry this won't pass any such law, but maybe there aren't.

Reply:
In the Haskell implementation, if there is a monad, then `select == selectM`:

```haskell
selectM :: Monad f => f (Either a b) -> f (a -> b) -> f b
selectM mx mf = do
    x <- mx
    case x of
        Left  a -> fmap ($a) mf
        Right b -> pure b
```

I haven't done the algebra, but they say that then implies `apS == <$>`:

```haskell
apS :: Selective f => f (a -> b) -> f a -> f b
apS f x = select (Left <$> f) ((&) <$> x)
```

I don't see any such laws for non-monads.

We are of course free to explore our own.  I'm just using the Haskell implementation as a starting point to see if this is even feasible in Cats 2.

Reply:
I'm +1 for adding those to MonadLaws.